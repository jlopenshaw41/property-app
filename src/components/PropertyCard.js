import React from "react";
import PropTypes from "prop-types";

const PropertyCard = (props) => {
  const { title, type, bedrooms, bathrooms, price, city, email } = props;
  return (
    <div className="property-card">
      <div className="property-card-title">{title}</div>
      <div className="property-card-type">{type}</div>
      <div
        className="property-card-bedrooms"
        data-testid="property-card-bedrooms"
      >
        {bedrooms}
      </div>
      <div
        className="property-card-bathrooms"
        data-testid="property-card-bathrooms"
      >
        {bathrooms}
      </div>
      <div className="property-card-price" data-testid="property-card-price">
        {price}
      </div>
      <div className="property-card-city">{city}</div>
      <div className="property-card-email">
        <a href={`mailto:${email}`}>{email}</a>
      </div>
    </div>
  );
};

PropertyCard.propTypes = {
  title: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  bedrooms: PropTypes.number.isRequired,
  bathrooms: PropTypes.number.isRequired,
  price: PropTypes.number.isRequired,
  city: PropTypes.string.isRequired,
  email: PropTypes.string.isRequired,
};

export default PropertyCard;
