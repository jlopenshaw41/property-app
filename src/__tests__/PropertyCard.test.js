import React from "react";
import { render } from "@testing-library/react";
import PropertyCard from "../components/PropertyCard";

describe("PropertyCard", () => {
  const validProps = {
    title: "Example property text",
    type: "Cottage",
    bathrooms: 2,
    bedrooms: 3,
    price: 300000,
    city: "Manchester",
    email: "email@address.com",
  };

  it("renders as expected", () => {
    const { asFragment } = render(
      <PropertyCard
        title={validProps.title}
        type={validProps.type}
        bathrooms={validProps.bathrooms}
        bedrooms={validProps.bedrooms}
        price={validProps.price}
        city={validProps.city}
        email={validProps.email}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("displays property title correctly", () => {
    const { getByText } = render(
      <PropertyCard
        title={validProps.title}
        type={validProps.type}
        bathrooms={validProps.bathrooms}
        bedrooms={validProps.bedrooms}
        price={validProps.price}
        city={validProps.city}
        email={validProps.email}
      />
    );
    expect(getByText(/Example property text/).textContent).toBe(
      "Example property text"
    );
  });

  it("displays type correctly", () => {
    const { getByText } = render(
      <PropertyCard
        title={validProps.title}
        type={validProps.type}
        bathrooms={validProps.bathrooms}
        bedrooms={validProps.bedrooms}
        price={validProps.price}
        city={validProps.city}
        email={validProps.email}
      />
    );
    expect(getByText(/Cottage/).textContent).toBe("Cottage");
  });

  it("displays bathrooms correctly", () => {
    const { getByTestId } = render(
      <PropertyCard
        title={validProps.title}
        type={validProps.type}
        bathrooms={validProps.bathrooms}
        bedrooms={validProps.bedrooms}
        price={validProps.price}
        city={validProps.city}
        email={validProps.email}
      />
    );
    expect(getByTestId("property-card-bathrooms").textContent).toBe("2");
  });

  it("displays bedrooms correctly", () => {
    const { getByTestId } = render(
      <PropertyCard
        title={validProps.title}
        type={validProps.type}
        bathrooms={validProps.bathrooms}
        bedrooms={validProps.bedrooms}
        price={validProps.price}
        city={validProps.city}
        email={validProps.email}
      />
    );
    expect(getByTestId("property-card-bedrooms").textContent).toBe("3");
  });

  it("displays price correctly", () => {
    const { getByTestId } = render(
      <PropertyCard
        title={validProps.title}
        type={validProps.type}
        bathrooms={validProps.bathrooms}
        bedrooms={validProps.bedrooms}
        price={validProps.price}
        city={validProps.city}
        email={validProps.email}
      />
    );
    expect(getByTestId("property-card-price").textContent).toBe("300000");
  });

  it("displays city correctly", () => {
    const { getByText } = render(
      <PropertyCard
        title={validProps.title}
        type={validProps.type}
        bathrooms={validProps.bathrooms}
        bedrooms={validProps.bedrooms}
        price={validProps.price}
        city={validProps.city}
        email={validProps.email}
      />
    );
    expect(getByText(/Manchester/).textContent).toBe("Manchester");
  });

  it("displays email correctly", () => {
    const { getByText } = render(
      <PropertyCard
        title={validProps.title}
        type={validProps.type}
        bathrooms={validProps.bathrooms}
        bedrooms={validProps.bedrooms}
        price={validProps.price}
        city={validProps.city}
        email={validProps.email}
      />
    );
    expect(getByText(/email@address.com/).textContent).toBe(
      "email@address.com"
    );
  });
});
