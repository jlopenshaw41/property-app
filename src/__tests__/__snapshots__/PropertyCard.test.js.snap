// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyCard renders as expected 1`] = `
<DocumentFragment>
  <div
    class="property-card"
  >
    <div
      class="property-card-title"
    >
      Example property text
    </div>
    <div
      class="property-card-type"
    >
      Cottage
    </div>
    <div
      class="property-card-bedrooms"
      data-testid="property-card-bedrooms"
    >
      3
    </div>
    <div
      class="property-card-bathrooms"
      data-testid="property-card-bathrooms"
    >
      2
    </div>
    <div
      class="property-card-price"
      data-testid="property-card-price"
    >
      300000
    </div>
    <div
      class="property-card-city"
    >
      Manchester
    </div>
    <div
      class="property-card-email"
    >
      email@address.com
    </div>
  </div>
</DocumentFragment>
`;
